### PR TITLE
Fix a crash when a process execution crashes and emits no data.

### LINKED ISSUE #265: Plot script crashes if a benchmark crashed.
I set one pexec's results (wct, and performance counters) to the empty list to emulate a crashed benchmark:

```
$ python -mpdb ~/research/warmup_experiment/bin/plot_krun_results xxx_results.json.bz2 -b "bencher3:binarytrees:Hotspot:default-java:0" -b "bencher3:binarytrees:Hotspot:default-java:1" -o mis.pdf
[2] > /home/vext01/research/warmup_experiment/bin/plot_krun_results(4)<module>()
-> """
(Pdb++) c                                                                                                /home/vext01/.local/lib/python2.7/site-packages/matplotlib/__init__.py:878: UserWarning: axes.color_cycle is deprecated and replaced with axes.prop_cycle; please use the latter.
  warnings.warn(self.msg_depr % (key, alt_key))
Saving results to: mis.pdf
Using matplotlib version 1.5.3 with backend agg
Charting with sliding window size: 200, xlimits: None
Loading: xxx_results.json.bz2
Collecting instrumentation data for xxx_results.json.bz2 from xxx_instr_data.
WARNING: Skipping: binarytrees:Hotspot:default-java from bencher3 (benchmark crashed)
Saved: mis.pdf
Traceback (most recent call last):
  File "/usr/lib/python2.7/pdb.py", line 1314, in main
    pdb._runscript(mainpyfile)
  File "/usr/lib/python2.7/pdb.py", line 1233, in _runscript
    self.run(statement)
  File "/usr/lib/python2.7/bdb.py", line 400, in run
    exec cmd in globals, locals
  File "<string>", line 1, in <module>
  File "/home/vext01/research/warmup_experiment/bin/plot_krun_results", line 4, in <module>
    """
  File "/home/vext01/research/warmup_experiment/bin/plot_krun_results", line 240, in main
    bmark, vm, mc = all_subplot_titles[index][0].split(', ')[:3]
IndexError: list index out of range
Uncaught exception. Entering post mortem debugging
Running 'cont' or 'step' will restart the program
[5] > /home/vext01/research/warmup_experiment/bin/plot_krun_results(240)main()
-> bmark, vm, mc = all_subplot_titles[index][0].split(', ')[:3]
(Pdb++) bt                                                                                               
[0]   /usr/lib/python2.7/pdb.py(1314)main()
-> pdb._runscript(mainpyfile)
[1]   /usr/lib/python2.7/pdb.py(1233)_runscript()
-> self.run(statement)
[2]   /usr/lib/python2.7/bdb.py(400)run()
-> exec cmd in globals, locals
[3]   <string>(1)<module>()
[4]   /home/vext01/research/warmup_experiment/bin/plot_krun_results(4)<module>()
-> """
[5] > /home/vext01/research/warmup_experiment/bin/plot_krun_results(240)main()
-> bmark, vm, mc = all_subplot_titles[index][0].split(', ')[:3]
(Pdb++) all_subplot_titles                                                                               
[[]]
```

### PR DESCRIPTION
**Pull request by vext01**
Fixes #265.

I hacked some data to make it look like pexec 0 for binary trees, hotspot crashed. The script now skips this pexec and keeps the correct titles.

g

### CODE REVIEW

File: bin/plot_krun_results [hunk 1]
```diff
@@ -250,19 +250,46 @@ def main(is_interactive, data_dcts, plot_titles, window_size, outfile,
...
-                                   all_changepoint_means[index], median, tukey,
-                                   inset_xlimit, legend_off, core_cycles, cycles_ylimits)
-            if not is_interactive:
-                fig.set_size_inches(*export_size)
-                pdf.savefig(fig, dpi=fig.dpi, orientation='landscape',
-                            bbox_inches='tight')
-                pyplot.close()
+                  (mc, bmark, vm, index + 1, len(pages))
+
+            # strip out indicies where the benchmark crashed
```
Comment (snim2) [inline bin/plot_krun_results]: `# Strip out indices where the benchmark crashed.`

Comment (vext01) [reply bin/plot_krun_results]: caps and full stop?

File: bin/plot_krun_results [hunk 2]
```diff
@@ -250,19 +250,46 @@ def main(is_interactive, data_dcts, plot_titles, window_size, outfile,
...
+            def only_uncrashed(data):
```
Comment (snim2) [inline bin/plot_krun_results]: I think this could be factored out, rather than having a nested function.

File: bin/plot_krun_results [hunk 3]
```diff
@@ -250,19 +250,46 @@ def main(is_interactive, data_dcts, plot_titles, window_size, outfile,
...
+                if data is None:
+                    return None
+                ret = []
```
Comment (snim2) [inline bin/plot_krun_results]: `ret = list()` as elsewhere in the file

File: bin/plot_krun_results [hunk 4]
```diff
@@ -250,19 +250,46 @@ def main(is_interactive, data_dcts, plot_titles, window_size, outfile,
...
+                for i in xrange(len(page)):
+                    if page[i]:
+                        ret.append(data[i])
+                    else:
+                        if data == page:  # stops repeated printing of warning
+                            print("WARNING: requested pexec crashed: "
```
Comment (snim2) [inline bin/plot_krun_results]: Likewise, quotes are single quotes

**Comment by snim2** [comment]
OK, what happens if you use `-b` and include the crashed benchmark?

File: bin/plot_krun_results (hunk 2 above)
Comment (vext01) [reply bin/plot_krun_results]: it makes the code shorter, as it is a closure over page

File: bin/plot_krun_results (hunk 3 above)
Comment (vext01) [reply bin/plot_krun_results]: Why do we do this by the way? `list()` is longer to type and Python has syntax for lists.

**Comment by vext01** [comment]
If you use -b and only the crashed benchmark, you are warned twice:
 * once when it notices you asked for a crashed pexec
 * again when the resulting page is empty.

File: bin/plot_krun_results [hunk 5]
```diff
@@ -250,19 +250,46 @@ def main(is_interactive, data_dcts, plot_titles, window_size, outfile,
...
-                                   all_changepoint_means[index], median, tukey,
-                                   inset_xlimit, legend_off, core_cycles, cycles_ylimits)
-            if not is_interactive:
-                fig.set_size_inches(*export_size)
-                pdf.savefig(fig, dpi=fig.dpi, orientation='landscape',
-                            bbox_inches='tight')
-                pyplot.close()
+                  (mc, bmark, vm, index + 1, len(pages))
+
+            # Strip out indicies where the benchmark crashed.
```
Comment (snim2) [inline bin/plot_krun_results]: s/indicies/indices/g

Comment (vext01) [reply bin/plot_krun_results]: ohhhhh

**Comment by snim2** [comment]
OK, this looks ready to squash

**Comment by vext01** [comment]
splat

**Review by snim2 — APPROVED** [review]

**Pull request merged**